xclim/indices/_agro.py

```diff
@@ -951,80 +953,63 @@ def standardized_precipitation_index(
             f"The method `{method}` is not supported for distribution `{dist}`."
         )
 
-    # calibration period
-    cal_period = pr_cal.time[[0, -1]].dt.strftime("%Y-%m-%dT%H:%M:%S").values.tolist()
+    # freq is None allows to use monthly precips
+    if freq is None:
+        group = "time.month"
```

Comment:
Probably it would be best to allow for any grouper type. group should be inferred from data in this case